### PR TITLE
Add tests for image and null space for GPU

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Comment by kshyatt** [comment]
This will need #80 to merge first

**Comment by kshyatt** [comment]
@lkdvos I had to modify the `truncate` a little to play nicely with 1.10 and AMDGPU -- the result (and the number of allocs) should be similar, but GPUCompiler really didn't like the logical indexing and this was least ugly way I found to set this up...

File: ext/MatrixAlgebraKitAMDGPUExt/MatrixAlgebraKitAMDGPUExt.jl [hunk 1]
```diff
@@ -154,4 +154,15 @@ function MatrixAlgebraKit._avgdiff!(A::StridedROCMatrix, B::StridedROCMatrix)
...
 end
 
+function MatrixAlgebraKit.truncate(::typeof(MatrixAlgebraKit.left_null!), US::Tuple{TU, TS}, strategy::MatrixAlgebraKit.TruncationStrategy) where {TU <: ROCArray, TS}
+    # TODO: avoid allocation?
+    U, S = US
+    extended_S = vcat(diagview(S), zeros(eltype(S), max(0, size(S, 1) - size(S, 2))))
+    ind = MatrixAlgebraKit.findtruncated(extended_S, strategy)
+    trunc_cols = collect(1:size(U, 2))[ind]
+    Utrunc = similar(U, (size(U, 1), length(trunc_cols)))
+    Utrunc .= U[:, trunc_cols]
```
[the remark this answers is not in the document]
Comment (kshyatt) [reply ext/MatrixAlgebraKitAMDGPUExt/MatrixAlgebraKitAMDGPUExt.jl]: Let me see!

File: test/amd/orthnull.jl [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+end
+function MatrixAlgebraKit.initialize_output(
+        ::typeof(svd_compact!), A::LinearMap,
+        alg::GPU_SVDAlgorithm
+    )
+    return LinearMap.(initialize_output(svd_compact!, parent(A), alg))
+end
+function MatrixAlgebraKit.svd_compact!(A::LinearMap, USVᴴ, alg::GPU_SVDAlgorithm)
+    return LinearMap.(svd_compact!(parent(A), parent.(USVᴴ), alg))
+end
```
[the remark this answers is not in the document]
Comment (kshyatt) [reply test/amd/orthnull.jl]: Yes can do

File: ext/MatrixAlgebraKitAMDGPUExt/MatrixAlgebraKitAMDGPUExt.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (kshyatt) [reply ext/MatrixAlgebraKitAMDGPUExt/MatrixAlgebraKitAMDGPUExt.jl]: Nice, it does work...
